server/src/route/cluster_based.rs

```diff
@@ -19,8 +26,58 @@ impl ClusterBasedRouter {
     }
 }
 
+// Parse the raw endpoint which should be the form: <domain_name>:<port>
+//
+// Returns `None` if fail to parse.
+fn try_parse_endpoint(raw: &str) -> Option<Endpoint> {
```

Comment:
standard library has this trait https://doc.rust-lang.org/std/str/trait.FromStr.html#tymethod.from_str (which provides `.parse()`)

Reply:
I have tried, but I find that the Endpoint is defined in the ceresdbproto crate and I can't impl `FromStr` for it.

Reply:
I have tried, but I find that the Endpoint is defined in the ceresdbproto crate and I can't impl `FromStr` for it.

Reply:
I have tried, but I find that the Endpoint is defined in the ceresdbproto crate and I can't impl `FromStr` for it.

Reply:
I have tried, but I find that the Endpoint is defined in the ceresdbproto crate and I can't impl `FromStr` for it.

Reply:
I have a suggestion: redefine `Endpoint` in our lib and operate on it, and implement methods like `to_pb()` and `from_pb()`. By doing this https://github.com/CeresDB/ceresdb/pull/215#discussion_r954466140 can also be solved.